### PR TITLE
ENH: merge methods are now variadic

### LINKED ISSUE #103: Merging capabilities should not be limited to two objects at a time
With `qiime feature-table merge`, users should be able to merge an arbitrary number of tables at a time. Currently the interface only allows for two tables to be merged at a time, which leads to a proliferation of intermediate tables. The same thing applies for `qiime feature-table merge-seq-data`.

### PR DESCRIPTION
**Pull request by ebolyen**
fixes #103

### CODE REVIEW

File: q2_feature_table/plugin_setup.py [hunk 1]
```diff
@@ -154,78 +154,67 @@
...
-                        'containing all feature sequences from data1 and '
-                        'data2.')
+                        'containing all feature sequences provided.')
     },
-    name="Combine two collections of feature sequences",
-    description="Combines a pair of feature data objects which may or may not "
+    name="Combine collections of feature sequences",
+    description="Combines feature data objects which may or may not "
                 "contain data for the same features. If different feature "
                 "data is present for the same feature id in the two inputs, "
```
Comment (thermokarst) [inline q2_feature_table/plugin_setup.py]: `s/two//`

> data is present for the same feature id in the ~two~ inputs,

File: q2_feature_table/plugin_setup.py [hunk 2]
```diff
@@ -154,78 +154,67 @@
...
         'merged_data': ('The resulting collection of feature taxonomies '
-                        'containing all feature taxonomies from data1 and '
-                        'data2.')
+                        'containing all feature taxonomies provided.')
     },
-    name="Combine two collections of feature taxonomies",
+    name="Combine collections of feature taxonomies",
     description="Combines a pair of feature data objects which may or may not "
                 "contain data for the same features. If different feature "
                 "data is present for the same feature id in the two inputs, "
```
Comment (thermokarst) [inline q2_feature_table/plugin_setup.py]: `s/two//`

> data is present for the same feature id in the ~two~ inputs,

File: q2_feature_table/plugin_setup.py [hunk 3]
```diff
@@ -154,78 +154,67 @@
...
+    inputs={'tables': List[FeatureTable[Frequency]]},
     parameters={
         'overlap_method': Str % Choices(q2_feature_table.overlap_methods()),
     },
     outputs=[
         ('merged_table', FeatureTable[Frequency])],
     input_descriptions={
-        'table1': 'The first feature table to be merged.',
-        'table2': 'The second feature table to be merged.',
+        'tables': 'The tables to be merged together.',
```
Comment (thermokarst) [inline q2_feature_table/plugin_setup.py]: Should we emulate the `metadata` merging language here?

e.g. `The tables to be merged together. This option may be specified multiple times.`, or something along those lines?

The same question applies for the other two merge methods impacted here.

File: q2_feature_table/plugin_setup.py [hunk 4]
```diff
@@ -154,78 +154,67 @@
...
+                        'containing all feature taxonomies provided.')
     },
-    name="Combine two collections of feature taxonomies",
+    name="Combine collections of feature taxonomies",
     description="Combines a pair of feature data objects which may or may not "
                 "contain data for the same features. If different feature "
                 "data is present for the same feature id in the two inputs, "
-                "the data from the first (data1) will be propagated to the "
-                "result."
+                "the data from the first  will be propagated to the result."
```
Comment (thermokarst) [inline q2_feature_table/plugin_setup.py]: There are two spaces between `first` and `will`.

File: q2_feature_table/_merge.py [hunk 5]
```diff
@@ -8,53 +8,59 @@
...
     return ('error_on_overlapping_sample', 'error_on_overlapping_feature',
             'sum')
 
 
-def _get_overlapping(table1, table2, axis):
-    table1_ids = set(table1.ids(axis=axis))
-    table2_ids = set(table2.ids(axis=axis))
-    return table1_ids & table2_ids
+def _get_overlapping(tables, axis):
+    ids = collections.Counter()
```
Comment (thermokarst) [inline q2_feature_table/_merge.py]: Nice!

File: q2_feature_table/plugin_setup.py (hunk 3 above)
Comment (ebolyen) [reply q2_feature_table/plugin_setup.py]: That's kind of an implementation detail of the CLI, so I don't know if it makes sense here.

Comment (thermokarst) [reply q2_feature_table/plugin_setup.py]: 👍 I agree, but it does seem hard to make the language general enough in an interface. I am cool either way.

Comment (ebolyen) [reply q2_feature_table/plugin_setup.py]: What about `Multiple tables to be merged together` or something like that?

**Comment by ebolyen** [comment]
@thermokarst, I matched the wording of `merge-*` for `merge`, those sounded better.

**Pull request merged**